### DISCUSSION TITLE
Modify some part to fit my need

### DISCUSSION THREAD

**Discussion by fleixi** [photoview/photoview#250] [General]
Now that the software is working i like to modify some part to better fit my needs. So the question is where to start for this goals:

- Translate webpage to german (maybe in a way i could give it back to your project to)
- default startingpoint to album not timeline
- disable facedetection (nice to have but i would possible overtaxe my people (grandmother and so on))
- https via nginx

for the last i already think i could disable PHOTOVIEW_SERVE_UI but than i need to know how i need to serve the files via nginx correctly. Is there some working example somewhere?

**Comment by viktorstrate** [comment]
Great questions:

> - Translate webpage to german (maybe in a way i could give it back to your project to)

There is an open issue on this one (#243), it requires some work to set up, but it's something that I'd like to work on in the near future.

> - default startingpoint to album not timeline

It would be relatively easy to manually make this change, though then future updates would need to be patched manually each time.
I'd like to add a settings page for non-admin users, where it should be possible to reset your own password, and maybe customizations on a per user basis, like dark mode when that gets implemented, maybe it could be a nice place to add an option for this as well?

> - disable facedetection (nice to have but i would possible overtaxe my people (grandmother and so on))

This could maybe be implemented in a relatively simple way by adding a new environment variable to disable it on start up.
Something like `PHOTOVIEW_DISABLE_PHOTOVIEW=1` to disable.

> - https via nginx

The simplest solution is just to use SERVE_UI and then reverse proxy everything with nginx, then the photoview server would handle proper caching for efficiency etc.
There is no working example, but maybe something like this (note that I haven't tested it):

```
server {
    listen 80;
    server_name photos.qpqp.dk;

    location / {
        proxy_pass http://127.0.0.1:4001;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        access_log /var/log/nginx/photoview.access.log;
        error_log /var/log/nginx/photoview.error.log;
    }

}
```

For SSL, certbot can be used to automatically configure certificates and nginx configuration.
https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-20-04

---

It should be possible to only reverse proxy the API requests and then serve the static UI files with Nginx directly, though I haven't tested this in a while so there might be some issues with this, especially #184.

**Reply by fleixi** [reply]
I doesn't mind to change the parts manually at beginning i only need some hit where i code i need to search for the parts

Translation:

 a i missed this report. Until than are there some place in code where i could replace the most imported parts manually?

Default startpoint:

this would be a good point to place this option. Until it is implemented: Where could i change this manually.

About changing your own password: Please implement this in a way it could be disabled. I like to share some accounts across family and if everyone can change it it will be a mess.

Facedetection:

Same as above: Until there is a option where could i disable it in code? Just the path, i will find out howto on my self

Nginx:

thankyou this is very helpful. Setting up lets-encrypt in nginx is something i already did before but the correct proxysettings a always a futzing
